extensions/ext/GLSL_EXT_integer_dot_product.txt

```diff
@@ -0,0 +1,288 @@
+    -----------------
+
+    For informational purposes (non-normative), the following is an
+    expected way for an implementation to map GLSL constructs to SPIR-V
+    constructs, the SPIR-V capabilities they require, and the Vulkan
+    feature bits those capabilities require.
+
+    All functions introduced in this extension require support for
+    the shaderIntegerDotProduct Vulkan feature and DotProductKHR SPIR-V
+    capability.
+
+    The dotPacked4x8EXT and dotPacked4x8AccSatEXT functions require support for
+    the DotProductInput4x8BitPackedKHR SPIR-V capability.
+
+    All functions with ivec8 or uvec8 input vectors require support for the
+    DotProductInput4x8BitKHR SPIR-V capability.
+
+    All functions with input vectors of other types require support for the
+    DotProductInputAllKHR SPIR-V capability.
+
+    Function described in this extension map to SPIR-V instructions as follows:
+
+      dot(Packed4x8)EXT(unsigned, unsigned)                 -> Opcode: OpUDotKHR
+      dot(Packed4x8)EXT(signed, unsigned)                   -> Opcode: OpSUDotKHR
+      dot(Packed4x8)EXT(unsigned, signed)                   -> Opcode: OpSUDotKHR
+      dot(Packed4x8)EXT(signed, signed)                     -> Opcode: OpSDotKHR
+      dot(Packed4x8)AccSatEXT(unsigned, unsigned, unsigned) -> Opcode: OpUDotAccSatKHR
+      dot(Packed4x8)AccSatEXT(signed, unsigned, signed)     -> Opcode: OpSUDotAccSatKHR
+      dot(Packed4x8)AccSatEXT(unsigned, signed, signed)     -> Opcode: OpSUDotAccSatKHR
+      dot(Packed4x8)AccSatEXT(signed, signed, signed)       -> Opcode: OpSDotAccSatKHR
+
+Modifications to the OpenGL Shading Language Specification, Version 4.60
+
+    Including the following lines in a shader can be used to control the
+    language features described in this extension:
+
+      #extension GL_EXT_integer_dot_product : <behavior>
+
+    where <behavior> is as specified in section 3.3.
+
+    New preprocessor #defines are added to the OpenGL Shading Language:
+
+      #define GL_EXT_integer_dot_product            1
+
+    Modify Section 8.5, Geometric Functions
+
+    Add new rows to the table of built-in functions:
+
+      uint dotEXT(uvec2 a, uvec2 b)
+      int dotEXT(ivec2 a, ivec2 b)
+      int dotEXT(ivec2 a, uvec2 b)
+      int dotEXT(uvec2 a, ivec2 b)
+
+      uint dotEXT(uvec3 a, uvec3 b)
+      int dotEXT(ivec3 a, ivec3 b)
+      int dotEXT(ivec3 a, uvec3 b)
+      int dotEXT(uvec3 a, ivec3 b)
+
+      uint dotEXT(uvec4 a, uvec4 b)
+      int dotEXT(ivec4 a, ivec4 b)
+      int dotEXT(ivec4 a, uvec4 b)
+      int dotEXT(uvec4 a, ivec4 b)
+
+      uint dotPacked4x8EXT(uint a, uint b)
+      int dotPacked4x8EXT(int a, uint b)
+      int dotPacked4x8EXT(uint a, int b)
+      int dotPacked4x8EXT(int a, int b)
+
+      If GL_EXT_shader_explicit_arithmetic_types_int8 is enabled:
+
+      uint dotEXT(u8vec2 a, u8vec2 b)
+      int dotEXT(i8vec2 a, u8vec2 b)
+      int dotEXT(u8vec2 a, i8vec2 b)
+      int dotEXT(i8vec2 a, i8vec2 b)
```

Comment:
It's a bit odd that 32 and 64 bit dot products are not extended but these lower-bit-depth variants extend to 32-bits first. SPIR-V has opcodes for 8 x 8 -> 8, etc. which may be faster than these 8x8 -> 32 variants. Do you think these are more useful to apps?